src/transformers/functional.js

```diff
@@ -0,0 +1,95 @@
+import propTypeToFlowType from '../helpers/propTypeToFlowType';
+
+function removeComponentAssignmentPropTypes(ast, j) {
+  const componentToPropTypesRemoved = {};
+
+  ast.find(j.AssignmentExpression, {
+    left: {
+      property: {
+        name: 'propTypes',
+      },
+    },
+  }).forEach(p => {
+    const objectName = p.value.left.object.name;
+    const properties = p.value.right.properties;
+    const flowTypesRemoved = properties.map(property => {
+      const t = propTypeToFlowType(j, property.key, property.value);
+      t.comments = property.comments;
+      return t;
+    });
+
+    componentToPropTypesRemoved[objectName] = flowTypesRemoved;
+  })
+  .remove();
+
+  return componentToPropTypesRemoved;
+}
+
+function insertTypeIdentifierInFunction(functionPath, j, typeIdentifier) {
+  const functionRoot = functionPath.value.init || functionPath.value;
+
+  const params = functionRoot.params;
+  const param = params[0];
+
+  const newTypeAnnotation = j.typeAnnotation(
+    j.genericTypeAnnotation(j.identifier(typeIdentifier), null)
+  );
+
+  if (param.type === 'Identifier') {
+    param.typeAnnotation = newTypeAnnotation;
+  } else if (param.type === 'ObjectPattern') {
+    // NOTE: something is wrong with recast and objectPatterns...
```

Comment:
Didn't have type to investigate this more thoroughly, but seems to be a limitation in recast.